indexmanager/registry.go

```diff
@@ -99,7 +100,7 @@ func (mr *mappingsRegistry) ParseIndexName(indexName string) *IndexName {
 	}
 
 	if documentKind == "" {
-		return nil
+		return nil, false
```

Comment:
maybe I'm missing some context here, but why not just keep the `nil` return and do a check for it when consuming this function?  it looks like we only ever return either `nil, false` or `(not nil), true`.

Reply:
That's what I had originally, but I thought maybe using the `value, ok` pattern would be a little more explicit. I'll switch it back.